src/Symfony/Component/HttpClient/TraceableHttpClient.php

```diff
@@ -31,15 +31,20 @@ final class TraceableHttpClient implements HttpClientInterface, ResetInterface,
     public function __construct(
         private HttpClientInterface $client,
         private ?Stopwatch $stopwatch = null,
+        public bool $enabled = true,
```

Comment:
Should the public property be internal or do we want to consider it part of our API covered by BC ?

Note that widening the type of a public property is a BC break (just like widening a return type as it means callers could get results they don't support), so making it a public API may prevent changing the type to `bool|(Closure(): bool)` in the future if that's the solution we want to use for the dynamic disabling.

If we want to be able to mark the property as internal, we probably need to use a normal property and a normal constructor argument instead of a promoted property.

Reply:
You're right. I updated the PR to account for this concern, and the other one also.